tests/heatbath_test.cpp

```diff
@@ -53,101 +52,153 @@ void display_test_info()
              dimPartitioned(3));
 }
 
-void heatbath_test(int argc, char **argv)
+// CURAND random generator initialization
```

Comment:
Not necessarily cuRAND (e.g., on non-CUDA platforms).  Also why the need for a `shared_ptr` here?

Reply:
https://github.com/lattice/quda/commit/2f32193a0015fdff402242a621bebabbb536c98d removes `shared_ptr` via adding a default constructor to `quda::RNG`